src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -0,0 +1,350 @@
+    const text = omnibox.getText();
+
+    return this.cache.getSuggestions(text, () => this.getFacetValueSuggestions(text, omnibox));
+  }
+
+  private getFacetValueSuggestions(text: string, omnibox: Omnibox): Promise<IOmniboxSuggestion[]> {
+    const wordsToQuery = this.options.useValueFromSearchbox ? [text] : [];
+
+    const promise: Promise<IOmniboxSuggestion[]> = this.getQuerySuggestionsKeywords(omnibox)
+      .then(suggestions => {
+        return this.listFieldValuesBatch(_.unique(wordsToQuery.concat(suggestions)));
+      })
+      .then((response: IFacetValueSuggestionsResponse) => {
+        return this.filterCurrentlySelectedValuesFromResults(response.responses, response.reference).map(resultToShow => {
+          return <IOmniboxSuggestion>{
+            html: this.buildDisplayNameForRow(resultToShow, omnibox),
+            onSelect: () => this.onRowSelection(resultToShow, omnibox)
+          };
+        });
+      })
+      .catch(() => []);
+    return promise;
+  }
+
+  private buildDisplayNameForRow(row: IFacetValueSuggestionRow, omnibox: Omnibox): string {
+    const keyword = DomUtils.highlightElement(row.keyword, omnibox.getText(), 'coveo-omnibox-hightlight2');
+    const facetValue = DomUtils.highlightElement(row.value, row.value, 'coveo-omnibox-hightlight');
+    const details = this.options.displayEstimateNumberOfResults ? ` (${row.numberOfResults} results)` : '';
+    return `${keyword} in ${facetValue}${details}`;
+  }
+
+  private filterCurrentlySelectedValuesFromResults(
+    fieldResponses: IFacetValueBatchResponse[],
+    fieldTotalReference: IFacetValueReference
+  ): IFacetValueSuggestionRow[] {
+    const currentSelectedValues: string[] = this.queryStateModel.get(this.queryStateFieldFacetId) || [];
+    const resultsWithoutSelectedValues: IFacetValueSuggestionRow[] = fieldResponses.reduce((allValues, fieldResponse) => {
+      // Remove suggestions that are already checked.
+      const filteredResults = fieldResponse.values.filter(result => !currentSelectedValues.some(value => value == result.value));
+
+      const suggestionRows = filteredResults.map(indexFieldValue => {
+        return <IFacetValueSuggestionRow>{
+          numberOfResults: indexFieldValue.numberOfResults,
+          keyword: fieldResponse.keyword,
+          value: indexFieldValue.value,
+          score: this.computeScoreForSuggestionRow(indexFieldValue, fieldTotalReference)
+        };
+      });
+      return allValues.concat(suggestionRows);
+    }, []);
+
+    this.logger.debug('FacetValue Suggestions Results', resultsWithoutSelectedValues);
+
+    const rankedResults = resultsWithoutSelectedValues.sort(
+      (a, b) => b.score.distanceFromTotalForField - a.score.distanceFromTotalForField
+    );
+    const preciseResults = rankedResults.splice(0, Math.ceil(this.options.numberOfSuggestions / 2));
+    const broadResults = rankedResults.slice(-1, Math.floor(this.options.numberOfSuggestions / 2));
```

Comment:
I am kind of experimenting with the score right now.

Since I can't determine which is best, this latest idea is to include half of the results that are precise (say 2/13000) and the other half broad (say 12000/13000).

Precise gives too many results that are not so useful (all 5 suggestions were 1 or 2 out of 13000)

Broad gives too much priority to the keyword typed in the searchbox. This keyword will always have better scoring since it always gives more results.

Thus this approach to benefit from both types :)